File: src/剑指Offer25.he-bing-liang-ge-pai-xu-de-lian-biao-lcof.100286/solution.ts
class ListNode {
    val: number
    next: ListNode | null
    constructor (val?: number, next?: ListNode | null) {
        this.val = (val === undefined ? 0 : val);
        this.next = (next === undefined ? null : next);
    }
}

function mergeTwoLists (l1: ListNode | null, l2: ListNode | null): ListNode | null {
    const dummyHead:ListNode = new ListNode(0);
    let last = dummyHead;
    while (l1 !== null && l2 !== null) {
        if (l1.val < l2.val) {
            last.next = l1;
            last = last.next;
            l1 = l1.next;
            last.next = null;
        } else {
            last.next = l2;
            last = last.next;
            l2 = l2.next;
            last.next = null;
        }
    }
    if (l1 !== null) {
        last.next = l1;
    } else if (l2 !== null) {
        last.next = l2;
    }
    return dummyHead.next;
}

File: src/剑指Offer25.he-bing-liang-ge-pai-xu-de-lian-biao-lcof.100286/solution.go
package lcof100286

type ListNode struct {
	Val  int
	Next *ListNode
}

func mergeTwoLists(l1 *ListNode, l2 *ListNode) *ListNode {
	dummyHead := ListNode{}
	last := &dummyHead

	for l1 != nil && l2 != nil {
		if l1.Val < l2.Val {
			n := l1.Next
			last.Next = l1
			l1.Next = nil
			last = last.Next
			l1 = n
		} else {
			n := l2.Next
			last.Next = l2
			l2.Next = nil
			last = last.Next
			l2 = n
		}
	}
	if l1 != nil {
		last.Next = l1
	}
	if l2 != nil {
		last.Next = l2
	}
	return dummyHead.Next
}
